### PR TITLE
migrated Migrations.EPiServer proj to net5

### PR DESCRIPTION
**Pull request by mmatras**
migrated Migrations.EpiServer and Migrations.Tests proj to .net 5

### CODE REVIEW

File: Migrations.EPiServer/Migrations.EPiServer.csproj [hunk 1]
```diff
@@ -1,171 +1,21 @@
...
-  </ItemGroup>
-  <ItemGroup>
-    <None Include="packages.config" />
+    <Content Include="Migrations.EPiServer.nuspec" />
   </ItemGroup>
   <ItemGroup>
-    <Content Include="Migrations.EPiServer.nuspec" />
+    <PackageReference Include="EPiServer.CMS.AspNetCore" Version="12.2.0" />
+    <PackageReference Include="EPiServer.Framework.AspNetCore" Version="12.2.0" />
+    <PackageReference Include="StructureMap" Version="4.5.2" />
```
Comment (tomaszmadeyski) [inline Migrations.EPiServer/Migrations.EPiServer.csproj]: Do we still need StructureMap in .net core??

File: Migrations.EPiServer/Migrations.EPiServer.csproj [hunk 2]
```diff
@@ -1,171 +1,21 @@
...
-      <Name>Migrations</Name>
-    </ProjectReference>
-  </ItemGroup>
-  <ItemGroup>
-    <None Include="packages.config" />
+    <Content Include="Migrations.EPiServer.nuspec" />
   </ItemGroup>
   <ItemGroup>
-    <Content Include="Migrations.EPiServer.nuspec" />
+    <PackageReference Include="EPiServer.CMS.AspNetCore" Version="12.2.0" />
```
Comment (tomaszmadeyski) [inline Migrations.EPiServer/Migrations.EPiServer.csproj]: I think it's better to set a range of versions, ie `Version="[12.0.0, 13)"` - this way we don't care what specific version of EpiServer is used

File: Migrations.EPiServer/MigrationsModule.cs [hunk 3]
```diff
@@ -7,7 +7,7 @@ namespace Forte.Migrations.EPiServer
     [ModuleDependency(typeof(global::EPiServer.Web.InitializationModule))]
     public class MigrationsModule : IInitializableModule
```
Comment (tomaszmadeyski) [inline Migrations.EPiServer/MigrationsModule.cs]: can we switch to .net's 5 `services.AddMigrations()` convetion rather than `[InitializableModule]` ?

Comment (tomaszmadeyski) [reply Migrations.EPiServer/MigrationsModule.cs]: and also, in this class I think we should no longer depend on `System.Configuration.ConfigurationManager` but maybe rather pass this flag (migrationsDisabled) as argument to `AddMigrations` method?

File: Migrations.EPiServer/DatabaseLockedSynchronizationContext.cs [hunk 4]
```diff
@@ -1,9 +1,11 @@
 ﻿using System;
 using System.Threading.Tasks;
 using EPiServer.Data;
+using EPiServer.ServiceLocation;
 
 namespace Forte.Migrations.EPiServer
 {
+    [ServiceConfiguration(ServiceType = typeof(IMigrationSynchronizationContext))]
```
Comment (tomaszmadeyski) [inline Migrations.EPiServer/DatabaseLockedSynchronizationContext.cs]: once we migrate to `.AddMigrations()` approach we can register this class in by calling `services.AddTransient<>`

File: Migrations.EPiServer/Migrations.EPiServer.csproj [hunk 5]
```diff
@@ -1,171 +1,21 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{18199349-9EB2-49E9-AC1F-6382AB8B8B93}</ProjectGuid>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (tomaszmadeyski) [inline Migrations.EPiServer/Migrations.EPiServer.csproj]: I wonder: will it work with .net6 ? Can we target both .net 5 and 6 at the same time? If not, maybe we should target .net6 instead as v5 has already reached end of life

File: Migrations.EPiServer/ServiceProviderActivator.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using EPiServer.ServiceLocation;
+using Microsoft.Extensions.DependencyInjection;
+using Org.BouncyCastle.Asn1.X509;
```
Comment (tomaszmadeyski) [inline Migrations.EPiServer/ServiceProviderActivator.cs]: I think we have redundant `using`s here

File: Migrations.EPiServer/Migrations.EPiServer.csproj (hunk 1 above)
Comment (mmatras) [reply Migrations.EPiServer/Migrations.EPiServer.csproj]: we don't, removed

File: Migrations.EPiServer/Migrations.EPiServer.csproj (hunk 2 above)
Comment (mmatras) [reply Migrations.EPiServer/Migrations.EPiServer.csproj]: fixed

File: Migrations.EPiServer/Migrations.EPiServer.csproj (hunk 5 above)
Comment (mmatras) [reply Migrations.EPiServer/Migrations.EPiServer.csproj]: hmm.. probably not, fixed

File: Migrations.EPiServer/ServiceProviderActivator.cs (hunk 6 above)
Comment (mmatras) [reply Migrations.EPiServer/ServiceProviderActivator.cs]: fixed

File: Migrations.EPiServer/MigrationsModuleExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        }
+        
+        public static IApplicationBuilder RunMigrations(this IApplicationBuilder builder, params Assembly[] assemblies)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            var serviceProvider = builder.ApplicationServices.GetRequiredService<IServiceProvider>();
```
Comment (tomaszmadeyski) [inline Migrations.EPiServer/MigrationsModuleExtensions.cs]: `builder.ApplicationServices` is of type `IServiceProvider` so I think we can simply do `var serviceProvider = builder.ApplicationServices;`, don't we?

**Review by tomaszmadeyski — APPROVED** [review]

**Pull request merged**